rb_tree.py

```diff
@@ -523,3 +534,100 @@ def _get_sibling(self, node):
             sibling = parent.right
             direction = 'R'
         return sibling, direction
+
+    def _build_tree_string(self, root=None, curr_index=0, index=False, delimiter='-'):
+    
```

Comment:
Could we also add the docstring from the original source?

Reply:
Ok, done!